pkg/sql/crdb_internal.go

```diff
@@ -337,6 +338,57 @@ CREATE TABLE crdb_internal.tables (
 	},
 }
 
+var crdbInternalTablesTableLastStats = virtualSchemaTable{
+	comment: "the latest stats for all tables accessible by current user in current database (KV scan)",
+	schema: `
+CREATE TABLE crdb_internal.table_row_stats (
+  table_id         INT         NOT NULL,
+  table_name       STRING      NOT NULL,
+  row_count        INT         NOT NULL
+)`,
+	generator: func(ctx context.Context, p *planner, dbContext *sqlbase.ImmutableDatabaseDescriptor) (virtualTableGenerator, cleanupFunc, error) {
+		row := make(tree.Datums, 3)
+		worker := func(pusher rowPusher) error {
+			return forEachTableDescAll(ctx, p, dbContext, virtualMany,
+				func(db *sqlbase.ImmutableDatabaseDescriptor, _ string, table *ImmutableTableDescriptor) error {
+					tableID := tree.DInt(table.ID)
+
+					query := `SELECT max("rowCount")` +
```

Comment:
hmm, the issue with doing it like this is that we have to make a query for each row, which is really unfortunate. I think that you could try to collect all of the latest stats up front into a map and then look up into that during this iteration. However, that has a problem with memory -- if the set of tables is too big then we'll OOM accessing this map... I think that would be unlikely however since all we'd be requesting are 3 `int`s per table.

Reply:
Yeah, good point. Fixed.